blockstm/condvar.go

```diff
@@ -0,0 +1,30 @@
+package blockstm
```

Comment:
feels like a lot of this logic that is top level should be in an `internal` package

Reply:
Yeah maybe. It currently will break a lot of things trying to do that and I'm not so concerned with people writing a bunch of libraries that will be broken by any changes we make here--it should be made clear that the blockstm code in this repo is subject to breaking changes more frequently than other parts of the sdk.